woocommerce/woocommerce-ios
# Product Sharing: Add analytics for AI-generated contents

Part of #9867 


## Description

This PR adds analytics for events related to AI-generated content of product sharing: 



Event | Trigger | Properties
-- | -- | --
*_product_sharing_ai_displayed | When the sheet for AI-generated share content is displayed. |  
*_product_sharing_ai_generate_tapped | When the “Write it for me” button is tapped on the AI product share sheet. | is_retry: true \| false
*_product_sharing_ai_share_tapped | When the “Share” button is tapped on the AI product share sheet. | with_message: true \| false
*_product_sharing_ai_dismissed | When the AI product share sheet is dismissed. |  
*_product_sharing_ai_message_generated | When a message is generated on the AI product share sheet. |  
*_product_sharing_ai_message_generation_failed | When message generation fails on the AI product share sheet. | error_context: $initiated-classname error_type: $errorType error_description: $errorDescription

This also added a check to identify whether or not the Generate button is tapped for a retry. This fixes the issue when the Generate button's title and image are updated when the message text editor is cleared.

Unit tests for `ShareProductCoordinator` will be added in a subsequent PR.

## Testing instructions

- Log in to a public WPCom site with at least one published product.
- Swipe-to-share the product in the product list or tap the Share button in it's detail screen.
- When the AI share sheet is displayed, notice in Xcode console: `🔵 Tracked product_sharing_ai_displayed`.
- Dismiss the sheet, an event should be tracked: `🔵 Tracked product_sharing_ai_dismissed`.
- Open the sheet again and tap "Write it for me". An event should be tracked: `🔵 Tracked product_sharing_ai_generate_tapped, properties: [AnyHashable("is_retry"): false, AnyHashable("blog_id"): <blog_id>, AnyHashable("is_wpcom_store"): true]`.
- After a message is generated, notice in Xcode console: `🔵 Tracked product_sharing_ai_message_generated, properties: [AnyHashable("blog_id"): <blog_id>, AnyHashable("is_wpcom_store"): true]`.
- Tap Regenerate, an event should be tracked: `🔵 Tracked product_sharing_ai_generate_tapped, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("is_retry"): true, AnyHashable("blog_id"): <blog_id>]`.
- Tap Share, an event should be tracked with the correct value for `with_message`: `🔵 Tracked product_sharing_ai_share_tapped, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): <blog_id>, AnyHashable("with_message"): true]`.
- With the AI share sheet still open, disconnect from the Internet and try generate a new message again. The request should fail and an event should be tracked with error properties: `🔵 Tracked product_sharing_ai_message_generation_failed`.

## Screenshots

N/A

---
